pkg/hub/submarineragent/controller.go

```diff
@@ -374,7 +374,7 @@ func (c *submarinerAgentController) syncSubmarinerConfig(ctx context.Context,
 	// prepare submariner cluster environment
 	errs := []error{}
 	var condition *metav1.Condition
-	if managedClusterInfo.Platform != "GCP" && managedClusterInfo.Platform != "OpenStack" {
+	if !isSpokePrepared(managedClusterInfo.Platform) {
```

Comment:
On second thought wrt `isSpokePrepared` as I suggested, for a provider not covered by cloud prepare, we won't set the `SubmarinerConfigConditionEnvPrepared` condition. By keeping the checks as they were then here we'll cover AWS and any other not handled by the spoke. For the latter, cloud prepare is a no-op but we will still set the condition here (the spoke will not). Next release we should move it all into the spoke as it's currently a bit confusing.